tests/pytorch_tests/function_tests/test_hessian_info_weights.py

```diff
@@ -0,0 +1,245 @@
+                                              granularity=hessian_common.HessianInfoGranularity.PER_OUTPUT_CHANNEL)
+            test_weights_hessian_trace_approx(hessian_service,
+                                              interest_point=ipt,
+                                              granularity=hessian_common.HessianInfoGranularity.PER_TENSOR)
+            test_weights_hessian_trace_approx(hessian_service,
+                                              interest_point=ipt,
+                                              granularity=hessian_common.HessianInfoGranularity.PER_ELEMENT)
+
+
+class WeightsHessianTraceAdvanceModelTest(BasePytorchTest):
+    def __init__(self, unit_test):
+        super().__init__(unit_test)
+        self.val_batch_size = 4
+
+    def create_inputs_shape(self):
+        return [[self.val_batch_size, 3, 32, 32]]
+
+    @staticmethod
+    def generate_inputs(input_shapes):
+        return generate_inputs(input_shapes)
+
+    def representative_data_gen(self):
+        input_shapes = self.create_inputs_shape()
+        yield self.generate_inputs(input_shapes)
+
+    def run_test(self, seed=0):
+        model_float = advanced_model()
+        pytorch_impl = PytorchImplementation()
+        graph = prepare_graph_with_configs(model_float, PytorchImplementation(), DEFAULT_PYTORCH_INFO,
+                                           self.representative_data_gen, generate_pytorch_tpc)
+        hessian_service = hessian_common.HessianInfoService(graph=graph,
+                                                            representative_dataset=self.representative_data_gen,
+                                                            fw_impl=pytorch_impl)
+        ipts = [n for n in graph.get_topo_sorted_nodes() if len(n.weights)>0]
+        for ipt in ipts:
+            test_weights_hessian_trace_approx(hessian_service,
+                                              interest_point=ipt,
+                                              num_scores=1,
+                                              granularity=hessian_common.HessianInfoGranularity.PER_OUTPUT_CHANNEL)
+            test_weights_hessian_trace_approx(hessian_service,
+                                              interest_point=ipt,
+                                              num_scores=2,
+                                              granularity=hessian_common.HessianInfoGranularity.PER_TENSOR)
+            test_weights_hessian_trace_approx(hessian_service,
+                                              interest_point=ipt,
+                                              num_scores=3,
+                                              granularity=hessian_common.HessianInfoGranularity.PER_ELEMENT)
+
+
+class WeightsHessianTraceMultipleOutputsModelTest(BasePytorchTest):
+    def __init__(self, unit_test):
+        super().__init__(unit_test)
+        self.val_batch_size = 1
+
+    def create_inputs_shape(self):
+        return [[self.val_batch_size, 3, 32, 32]]
```

Comment:
Same comment (for all tests): Please use the smallest needed input shape and models for unit tests.